frontend/src/components/LoginForm.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import styled from 'styled-components'
+
+import { user } from '../reducers/user'
+import { Profile } from './Profile'
+
+const LOGIN_URL = 'https://project-auth-cla-ellen.herokuapp.com/sessions'
+const SIGNUP_URL = 'https://project-auth-cla-ellen.herokuapp.com/users'
+
+export const LoginForm = () => {
+  const dispatch = useDispatch()
+  const accessToken = useSelector((store) => store.user.login.accessToken)
+  const statusMessage = useSelector((store) => store.user.login.statusMessage)
+      
+  const [name, setName] = useState('')
+  const [password, setPassword] = useState('')
+      
+  const handleLoginSuccess = (loginResponse) => {
+    dispatch(
+      user.actions.setAccessToken({ accessToken: loginResponse.accessToken })
+    )
+    dispatch(user.actions.setUserId({ userId: loginResponse.userId }))
+    dispatch(
+      user.actions.setStatusMessage({ statusMessage: 'Login success!' })
+    )
+    setName('')
+    setPassword('')
+  }
+      
+  const handleLoginFailed = (loginError) => {
+    dispatch(user.actions.setAccessToken({ accessToken: null }))
+    dispatch(user.actions.setStatusMessage({ statusMessage: loginError }))
+  }
+      
+  const handleSignup = (event) => {
+    event.preventDefault();
+          
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error('Sign up failed')
+        }
+          return res.json()
+      })
+        .then((json) => handleLoginSuccess(json))
+        .catch((err) => handleLoginFailed(err));
+  }
+      
+  const handleLogin = (event) => {
+    event.preventDefault()
+      
+    fetch(LOGIN_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error('Login failed. Please try to login again')
+        }
+          return res.json()
+      })
+        .then((json) => handleLoginSuccess(json))
+        .catch((err) => handleLoginFailed(err))
+  }
+      
+  if (accessToken) {
+    return (
+        <Profile />
+    )
+  }
```

Comment:
Two return in one component is not a good practice. In situations like this, you should use React Router. 🌌